griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,302 @@
+                    drive_service.files().update(fileId=doc_id, addParents=folder_id, fields="id, parents").execute()
+                else:
+                    return ErrorArtifact(f"Error: Folder not found for path {folder_path}")
+
+            if content:
+                save_content_params = {"document_id": doc_id, "content": content}
+                saved_document_id = self._save_to_doc(save_content_params)
+                return InfoArtifact(f"Content has been successfully saved to Google Doc with ID: {saved_document_id}.")
+            else:
+                return InfoArtifact(f"Google Doc '{file_path}' created with ID: {doc_id}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error creating/saving Google Doc: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to load content from memory and save it to a new Google Doc "
+                           "in the specified folder.",
+            "schema": Schema(
+                {
+                    "memory_name": str,
+                    "artifact_namespace": str,
+                    "file_name": str,
+                    Optional(
+                        "folder_path",
+                        description="Path of the folder where the file should be saved.",
+                        default=DEFAULT_FOLDER_PATH,
+                    ): str,
+                }
+            ),
+        }
+    )
+    def save_memory_artifacts_to_google_docs(self, params: dict) -> ErrorArtifact | InfoArtifact:
+        values = params["values"]
+        memory = self.find_input_memory(values["memory_name"])
+    
+        if memory:
+            artifacts = memory.load_artifacts(values["artifact_namespace"])
+    
+            if artifacts:
+                try:
+                    file_path = values["file_name"]
+                    content = "\n".join([a.value for a in artifacts])
+
+                    save_params = {
+                        "file_path": file_path,
+                        "content": content,
+                        "folder_path": values.get("folder_path", self.DEFAULT_FOLDER_PATH),
+                    }
+
+                    return self.save_content_to_google_doc(save_params)
+    
+                except Exception as e:
+                    return ErrorArtifact(f"Error: {e}")
+    
+            else:
+                return ErrorArtifact("No artifacts found.")
+        else:
+            return ErrorArtifact("Memory not found.")
+
+    @activity(
+        config={
+            "description": "Can be used to download multiple Google Docs based on their paths.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_paths",
+                        description="Destination file paths in the POSIX format. "
+                        "For example, 'foo/bar/baz.txt, foo/bar/baz2.txt'",
+                    ): [str]
```

Comment:
Not familiar with this syntax. Should this be `list[str]`?

Reply:
Yes!